File: ChainJava/src/chain/employees/SportingDirector.java
package chain.employees;

import chain.requests.Request;

/**
 * Created by Karolina_2 on 2015-04-01.
 */
public class SportingDirector extends Employee {

    public SportingDirector(Employee employee){
        this.setReceiver(employee);
    }

    @Override
    public void receiveARequest(Request request){
        if(isAllowedToReceive(request, "TransferRequest")){
            System.out.println("Sporting director has received " + request.getClass().getSimpleName());
        }
        else{
            super.receiveARequest(request);
        }
    }
}

File: ZTP_Zadanie3/ZTP_Zadanie3/chain/employees/SportingDirector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZTP_Zadanie3.chain.wnioski;

namespace ZTP_Zadanie3.pracownicy
{
    class SportingDirector : Employee
    {
        public SportingDirector(Employee employee)
        {
            this.setReceiver(employee);
        }

        public override void receiveARequest(Request request)
        {
            if (isAllowedToReceive(request, "TransferRequest"))
            {
                Console.WriteLine("Sporting director has received " + request.GetType().Name);
            }
            else
            {
                base.receiveARequest(request);
            }
        }
    }
}
